articles/changes.md

```diff
@@ -0,0 +1,245 @@
+
+
+### Unify duration and time types
+
+In ROS 1 the duration and time types are defined in the client libraries.
+The member names of the data structures are different in C++ (sec, nsec) and Python (secs, nsecs).
+
+In ROS 2 these types will be defines as messages and therefore be consistent across languages.
+
+
+### Remove sequence field from Header message
+
+The field has been deprecated for a long time and was not set consistently in ROS 1.
+
+
+## Client libraries
+
+### Across languages
+
+#### Notifications
+
+In ROS 1 all information about the ROS graph must be polled from the master.
+In ROS 2 changes will be published instead.
+
+
+#### Components with life cycle
+
+In ROS 1 every node usually has its own main function.
+In ROS 2 it will be recommended to subclass from a component which has a life cycle.
+
+The life cycle can be used by tools like roslaunch to start a system composed of many components in a deterministic way.
+
+
+#### Parameters and Dynamic Reconfigure
+
+In ROS 1 global parameters and node-specific dynamic reconfigure parameters are two separate concepts.
+In ROS 2 a unified approach is being used.
+It will be similar to dynamic reconfigure and a node named "global parameter server" will accept requests to set values unconditionally.
+In ROS 1 all this information needs to be polled for changes in ROS 2 changes will be published to notify other entities.
+
+
+#### Threading model
+
+In ROS 1 the developer can only choose between single-threaded execution or multi-threaded execution.
+In ROS 2 more granular execution models are available.
+
+
+#### Graph API
+
+In ROS 1 nodes and topics can be remapped at startup time only.
+In ROS 2 it will be possible to remap those at runtime as well as create aliases.
+
+
+### C and C++
+
+#### Support for real-time
+
+ROS 1 does not support writing real-time code but relies on external frameworks like Orocos.
+In ROS 2 it will be possible to write real-time nodes when using a proper RTOS and with carefully written user code.
+
+
+### C++
+
+#### Node vs. Nodelet
+
+In ROS 1 the API's for nodes and nodelets are different and requires the developer to decide the mapping of nodes to processes at programming time.
+In ROS 2 it will be possible to make this a deploy-time decision.
+It will be recommended to compile each component into a shared library which can then be loaded in a separate process or share the same process with other components (like ROS 1 nodelets).
+
+
+#### Allow multiple nodes per process
+
+In ROS 1 it is not possible to create more than one node in a process.
+This is due to the API itself but also because of internal implementation decisions.
+
+In ROS 2 it will be possible to create multiple nodes in a process.
+
+
+## Tools
+
+### roslaunch
+
+In ROS 1 roslaunch files are defined in xml with very limited capabilities.
+In ROS 2 launch file can be written in Python which enables to use more complex logic like conditionals etc.
```

Comment:
This topic is still poorly defined at this point. We don't have a design doc or a prototype, or even (in my opinion) a clear vision for how this will replace roslaunch (in terms of API, matching or dropping features of roslaunch). I think it would be good to mention that roslaunch is still under design and development as not to give people the idea that we've come to some concrete conclusion on the topic.